### PR TITLE
fix: workflow output path schedule mode

### LINKED ISSUE #719: Fix workflow output path when run in schedule mode
### Description

Output path should stay consistent in case of both manual and scheduled runs

References to blocks of code:
- Constant that configures the path format ([constants.py](https://github.com/atlanhq/application-sdk/blob/main/application_sdk/constants.py#L55))
- `build_output_path` method  ([utils.py](https://github.com/atlanhq/application-sdk/blob/main/application_sdk/activities/common/utils.py#L62))

### Reproduction Repo/Script (if any)

_No response_

### Reproduction Steps

```bash
1. Run a workflow in manual mode triggered via UI, observe the output path it will have this format - `artifacts/apps/redshift/workflows/atlan-redshift-1756943536/348b3af9-4d28-48fa-bc0d-eb70d6819ba2/`
2. Run a workflow in schedule mode, observe the output path (example) - `artifacts/apps/redshift/workflows/atlan-redshift-1756943536-2025-09-16T23:51:14Z/348b3af9-4d28-48fa-bc0d-eb70d6819ba2/`
```

### Logs / Tracebacks

```bash
N/A
```

### Expected vs Actual

Expected:
`artifacts/apps/redshift/workflows/atlan-redshift-1756943536/348b3af9-4d28-48fa-bc0d-eb70d6819ba2/`

Actual:
`artifacts/apps/redshift/workflows/atlan-redshift-1756943536-2025-09-16T23:51:14Z/348b3af9-4d28-48fa-bc0d-eb70d6819ba2/`

### Environment

None

### SDK Version

all

### PR DESCRIPTION
**Pull request by drockparashar**
### Changelog
-Ensures consistent workflow output path for both manual and scheduled runs. 
-Removes any timestamp or schedule-specific suffix from workflow_id when generating the output path. 
-Output path now always follows the format: artifacts/apps/{application_name}/workflows/{workflow_id}/{run_id}.

### Additional context (e.g. screenshots, logs, links)
image

closes #719 

### Checklist

- [ ] Additional tests added
- [ ] All CI checks passed
- [ ] Relevant documentation updated

---

### Copyleft License Compliance

- [ ] Have you used any code that is subject to a Copyleft license (e.g., GPL, AGPL, LGPL)?
- [ ] If yes, have you modified the code in the context of this project? please share additional details.

---

> [!NOTE]
> Sanitizes `workflow_id` in `build_output_path` by stripping schedule timestamp suffix and adds unit tests covering standard, scheduled, and empty IDs.
> 
> - **Utilities (`application_sdk/activities/common/utils.py`)**:
>   - **`build_output_path`**: Strip schedule timestamp suffix from `workflow_id` using `TIMESTAMP_PATTERN`; fallback to `"unknown-workflow"` if empty.
>   - Add compiled regex `TIMESTAMP_PATTERN` and required import.
> - **Tests (`tests/unit/activities/common/test_utils.py`)**:
>   - Add tests for `build_output_path` covering standard IDs, scheduled IDs with timestamps, and empty `workflow_id` fallback.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b257dcc0dec7d8d2ee325e119c5ce1bc0a5ebf04. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: application_sdk/activities/common/utils.py [hunk 1]
```diff
@@ -22,6 +23,9 @@
 
 logger = get_logger(__name__)
 
+# Compiled regex pattern for removing timestamp suffix from workflow IDs
+TIMESTAMP_PATTERN = re.compile(r"-\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$")
```
Comment (inishchith) [inline application_sdk/activities/common/utils.py:27]: let's keep this variable localized as it's only used by a a single method as of now.

File: application_sdk/activities/common/utils.py [hunk 2]
```diff
@@ -71,9 +76,19 @@ def build_output_path() -> str:
...
     """
+    # Sanitize workflow_id to remove any schedule/timestamp suffix
+    raw_workflow_id = get_workflow_id()
+
+    # Remove timestamp suffix (e.g., '-YYYY-MM-DDTHH:MM:SSZ') if present
+    sanitized_workflow_id = TIMESTAMP_PATTERN.sub("", raw_workflow_id)
+
+    # Fallback to raw workflow_id if sanitization results in empty string
+    if not sanitized_workflow_id:
+        sanitized_workflow_id = raw_workflow_id
```
Comment (inishchith) [inline application_sdk/activities/common/utils.py:87]: when could be this be the case? can you handle in tests if so

**Review by inishchith — COMMENTED** [review]
thanks for contributing. the change looks fine, could you contribute to unit tests for this function?

File: application_sdk/activities/common/utils.py (hunk 1 above)
Comment (drockparashar) [reply application_sdk/activities/common/utils.py:27]: I implemented that initially, but the copilot suggested me otherwise, what should i go with?

image

**Comment by drockparashar** [comment]
> thanks for contributing. the change looks fine, could you contribute to unit tests for this function?

Sure, I'll start working on it right away!

**Comment by inishchith** [comment]
@drockparashar - the tests are failing, you can validate this by running the unit tests locally.

**Comment by Aryamanz29** [comment]
Closing as part of v3 cleanup — this PR targets v2 code that has been superseded by the v3 architecture. The branch is preserved (not deleted) in case it's needed for reference.

If this is still relevant, please reopen against the current `main` branch (v3.0.x) and update the code to use the v3 APIs. See the [migration guide](https://github.com/atlanhq/application-sdk/blob/main/docs/upgrade-guide-v3.md) for v2 → v3 changes.
